Makefile

```
@@ -449,15 +429,21 @@ tests/%.prove-legacy: tests/%
 	    --no-provex --format-failures $(KPROVE_OPTS) --concrete-rules-file $(dir $@)concrete-rules.txt
 
 tests/specs/examples/erc20-spec/haskell/erc20-spec-kompiled/timestamp: tests/specs/examples/erc20-bin-runtime.k
-tests/specs/examples/erc20-bin-runtime.k: tests/specs/examples/ERC20.sol $(KEVM_LIB)/$(haskell_kompiled) $(kevm_pyk_includes) $(PYK_VENV)
-	$(ACTIVATE_PYK) && $(KEVM) solc-to-k $< ERC20 > $@
+tests/specs/examples/erc20-bin-runtime.k: tests/specs/examples/ERC20.sol $(KEVM_LIB)/$(haskell_kompiled) $(VENV)
+	$(MAKE) -C ./kevm_pyk venv-prod &&     \
+	. ./kevm_pyk/venv-prod/bin/activate && \
+	$(KEVM) solc-to-k $< ERC20 > $@
 
 tests/specs/examples/erc721-spec/haskell/erc721-spec-kompiled/timestamp: tests/specs/examples/erc721-bin-runtime.k
-tests/specs/examples/erc721-bin-runtime.k: tests/specs/examples/ERC721.sol $(KEVM_LIB)/$(haskell_kompiled) $(kevm_pyk_includes) $(PYK_VENV)
-	$(ACTIVATE_PYK) && $(KEVM) solc-to-k $< ERC721 > $@
-
-tests/specs/examples/empty-bin-runtime.k: tests/specs/examples/Empty.sol $(KEVM_LIB)/$(haskell_kompiled) $(kevm_pyk_includes) $(PYK_VENV)
-	$(ACTIVATE_PYK) && $(KEVM) solc-to-k $< Empty > $@
+tests/specs/examples/erc721-bin-runtime.k: tests/specs/examples/ERC721.sol $(KEVM_LIB)/$(haskell_kompiled) $(VENV)
+	$(MAKE) -C ./kevm_pyk venv-prod &&     \
+	. ./kevm_pyk/venv-prod/bin/activate && \
+	$(KEVM) solc-to-k $< ERC721 > $@
```

I'm pretty sure the first command doesn't need to be run in the same subshell as the later ones? So we should be able to do:

```suggestion
	$(MAKE) -C ./kevm_pyk venv-prod
	. ./kevm_pyk/venv-prod/bin/activate && $(KEVM) solc-to-k $< ERC721 > $@
```

---

Also, the first command could be it's own target, right?

So that each other target can depend on it directly, and make will only run it once?

---

Yes, that sounds right. I'll change this accordingly.